sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -254,3 +254,52 @@ case class Pow(left: Expression, right: Expression)
       """
   }
 }
+
+object Logarithm {
+  def apply(child: Expression): Expression = new Log(child)
+}
+
+case class Logarithm(left: Expression, right: Expression)
+  extends BinaryMathExpression((c1, c2) => math.log(c2) / math.log(c1), "LOG") {
```

Comment:
If we need to override the `eval()`, probably it's better to inherit from `BinaryArithmetic`, not `BinaryMathExpression`, then we needn't to pass the additional function object for its parent class.
Or can we remove the `eval()` from this class?